src/semgrep_agent/meta.py

```diff
@@ -276,25 +275,19 @@ def pr_title(self) -> Optional[str]:
         return os.getenv("CI_MERGE_REQUEST_TITLE")
 
 
-def detect_meta_environment() -> Type[GitMeta]:
+def generate_meta_from_environment(baseline_ref: Optional[str]) -> GitMeta:
     # https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables
     if os.getenv("GITHUB_ACTIONS") == "true":
-        return GithubMeta
-
-    # https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables
-    elif os.getenv("CIRCLECI") == "true":  # nosem
-        return GitMeta
-
-    # https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
-    elif os.getenv("TRAVIS") == "true":  # nosem
-        return GitMeta
+        return GithubMeta()
 
     # https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
     elif os.getenv("GITLAB_CI") == "true":
-        return GitlabMeta
-
-    elif os.getenv("CI"):  # nosem
-        return GitMeta
+        return GitlabMeta()
 
     else:  # nosem
-        return GitMeta
+        if not baseline_ref:
+            click.echo(
+                "Note that no baseline-ref was passed and detected environment is not Github or Gitlab to will default to perfoming a full scan. If you want semgrep to only report on new findings from a pull request please set baseline-ref ",
```

Comment:
Not sure what you meant this to say?

Reply:
maybe "is not GitHub or GitLab so semgrep-action will default to performing a full scan"